File: app/src/main/java/com/blackmirror/dongda/utils/ToastUtils.java
package com.blackmirror.dongda.utils;

import android.content.Context;
import android.widget.Toast;

public class ToastUtils {
    private static Toast toast;

    private ToastUtils() {
    }

    private static void getToast() {
        if (toast == null) {
            toast = Toast.makeText(AYApplication.getAppContext(),"",Toast.LENGTH_SHORT);
        }
    }

    public static void showShortToast(CharSequence msg) {
        showToast(AYApplication.getAppContext(), msg, Toast.LENGTH_SHORT);
    }

    public static void showShortToast(Context context,CharSequence msg) {
        showToast(context.getApplicationContext(), msg, Toast.LENGTH_SHORT);
    }

    public static void showShortToast(int resId) {
        showToast(AYApplication.getAppContext(), resId, Toast.LENGTH_SHORT);
    }

    public static void showLongToast(CharSequence msg) {
        showToast(AYApplication.getAppContext(), msg, Toast.LENGTH_SHORT);
    }

    private static void showToast(final Context context, final CharSequence msg, int duration) {
        try {
            getToast();
            toast.setText(msg);
            toast.setDuration(duration);
            toast.show();
        } catch (Exception e) {
            LogUtils.e(context.getClass(),"Exception==", e);
        }
    }

    private static void showToast(Context context, int resId, int duration) {
        try {
            if (resId == 0) {
                return;
            }
            getToast();
            toast.setText(resId);
            toast.setDuration(duration);
            toast.show();
        } catch (Exception e) {
            LogUtils.e(context.getClass(),"Exception==", e);
        }
    }
}

File: app/src/main/java/com/blackmirror/dongda/utils/ToastUtils.kt
package com.blackmirror.dongda.utils

import android.support.design.widget.Snackbar
import android.view.View
import android.widget.TextView
import android.widget.Toast
import com.blackmirror.dongda.R
import com.blackmirror.dongda.base.AYApplication

/**
 * Create By Ruge at 2018-12-06
 */
class SingltonToast private constructor(){

    private val toast:Toast

    init {
        toast= Toast.makeText(AYApplication.appContext,"",Toast.LENGTH_SHORT)
    }

    private object Holder {
        val singleton = SingltonToast()
    }

    companion object {
        val INSTANCE: SingltonToast by lazy { Holder.singleton }
    }

    fun getToast(): Toast {
        return toast
    }
}

fun showToast(msg: CharSequence,resId:Int=0,length:Int=Toast.LENGTH_SHORT){


    if (msg.isNullOrEmpty() && resId<=0){
        return
    }

    val toast = SingltonToast.INSTANCE.getToast()

    if (resId>0){
        toast.setText(resId)
    }else{
        toast.setText(msg)
    }

    toast.duration=length
    toast.show()
}

fun showSnackbar(view: View, content: String) {
    val snackbar = Snackbar.make(view, content, Snackbar.LENGTH_INDEFINITE)
    snackbar.setAction("关闭") { }
    snackbar.view.setBackgroundColor(AYApplication.appContext.resources.getColor(R.color.sys_bar_white))
    val v = snackbar.view
    (v.findViewById(R.id.snackbar_text) as TextView).setTextColor(
            AYApplication.appContext.resources.getColor(
                    R.color.text_black
            )
    )
    snackbar.show()
}